dockerfiles/grist/ressources/dinum-custom.js

```diff
@@ -0,0 +1,142 @@
+    // ASTUCE: Si la maintenance dure moins longtemps que prévu, réduire a postériori la valeur de cette variable de sorte à ce qu'elle n'apparaisse plus.
+    // Pas besoin de committer ensuite le changement dans le repo.
+    maintainanceDurationMinutes: 60,
+  };
+  // FIN CONFIGURER
+
+  const tz = { CEST: '0200', CET: '0100' };
+
+  const maintainanceStartDate = new Date(`${config.startDateWithoutTimezone}+${tz[config.selectedTz]}`);
+  const maintainanceEndDate = (function () {
+    const date = new Date(maintainanceStartDate); // If I may, Date API is really ackward, especially because it is not immutable
+    date.setMinutes(maintainanceStartDate.getMinutes() + config.maintainanceDurationMinutes);
+    return date;
+  })();
+
+
+  function showDialog() {
+    const dialog = document.createElement('dialog');
+    dialog.id = 'maintenancePopin';
+    const dateFormatter = new Intl.DateTimeFormat('fr-FR', {
+      dateStyle: 'full',
+      timeStyle: 'short',
+      timeZone: 'Europe/Paris',
+    });
+    const timeFormatter = new Intl.DateTimeFormat('fr-FR', {
+      timeStyle: 'short',
+      timeZone: 'Europe/Paris',
+    });
+
+    dialog.innerHTML = `
+        <p>
+        Une maintenance de Grist est prévue le ${dateFormatter.format(maintainanceStartDate)} heure de Paris
+          jusqu'à ${timeFormatter.format(maintainanceEndDate)}, période durant laquelle le service sera momentanément indisponible. <br>
+        Pour toute remarque ou question, merci de nous contacter via <a href="mailto:donnees@anct.gouv.fr">donnees@anct.gouv.fr</a></p>
+        <p>Merci de votre compréhension</p>
+        <form method="dialog">
+          <p>
+            <input type="checkbox" id="jaiCompris">
+            <label for="jaiCompris">J'ai lu et compris, merci de ne plus me montrer ce message</label>
+          </p>
+          <button disabled id="fermerPopinMaintenance">Fermer</button>
+        </form>
+    `;
+    document.body.appendChild(dialog);
+    document.getElementById('fermerPopinMaintenance').onclick = function onMessageUnderstood() {
+      localStorage.maintainanceStartDateAgreement = maintainanceStartDate.toISOString();
+    }
+    document.getElementById('jaiCompris').onchange = function (ev) {
+      document.getElementById('fermerPopinMaintenance').disabled = !ev.target.checked;
+    }
+    dialog.showModal();
+  }
+
+  if (Date.now() < maintainanceEndDate.getTime() &&
+    (!localStorage.maintainanceStartDateAgreement || localStorage.maintainanceStartDateAgreement !== maintainanceStartDate.toISOString())) {
+    window.addEventListener('load', showDialog);
+  }
+})();
+
+// ========================
+// END dialog
+// ========================
+
+// ========================
+// START gauffre
+// ========================
+
+window.addEventListener('load', (event) => {
+  const pathname = window.location.pathname;
+
+  // Gauffre is only inserted on grist home page
+  if (pathname !== '/o/docs/'){
```

Comment:
The home page may have other paths:
 - `/o/docs` without the trailing slash
 - `/o/myorg` with our without the trailing slash
 - `/o/docs/ws/...` or `myorg/ws/...` when navigating to some workspace

Sorry for having taking so long to figure that issue. I am thinking of a proposal

Reply:
OK, I hacked a bit, and found some way that is a bit… yeah… sportive…

Still wondering if there is no other and cleaner way to achieve this.

```js
function waitForElm(selector) {
  return new Promise(resolve => {
    if (document.querySelector(selector)) {
      return resolve(document.querySelector(selector));
    }

    const observer = new MutationObserver(mutations => {
      if (document.querySelector(selector)) {
        observer.disconnect();
        resolve(document.querySelector(selector));
      }
    });

    // If you get "parameter 1 is not of type 'Node'" error, see https://stackoverflow.com/a/77855838/492336
    observer.observe(document.body, {
      childList: false,
      subtree: false,
      attributes: true
    });
  });
}

window.addEventListener('load', async (ev) => {
    await waitForElm('body.interface-full');
    console.log('done waitForElm');
    // gristApp.topAppModel.appObs is not populated at that point, listen for the observer to their first change
    let listener; 
    listener = gristApp.topAppModel.appObs.addListener((appObs) => {
        console.log('pageType = ', appObs.pageType.get()); // logs either docs or home.
        // Should be disposed so we only listen for changes once, but failed to achieve doing that.
        // listener.dispose();
    });
});
```

Reply:
Ok,

I've made some changes based on your suggestion.